### PR TITLE
🔖 0.9.0 Cogs & Channel Edition [FULL]

### PR DESCRIPTION
**Pull request by Sigmanificient**
The `0.9.0` implements the cogs system!

*very original content i definitively wrote:*
### Basic example:

`run.py`

```py
>>> from pincer import Client
>>>
>>> class BotClient(Client):
...     def __init__(self, *args, **kwargs):
...         self.load_cog("cogs.say")
...         super().__init__(*args, **kwargs)
```

`cogs/say.py`

```py
>>> from pincer import command
>>>
>>> class SayCommand:
...     @command()
...     async def say(self, message: str) -> str:
...         return message
>>>
>>> setup = SayCommand
```

**Note on setup method**

The setup method may have 1 optional argument which contains the client.
The only restriction on the setup method is that it must be a callable, this means that the following code is also valid:

```py
def setup(client):
    return MyCog(client.bot.name)
```

## Channel Edition
*yes i even stole text from my previous issues!*

Implemented a `channel.edit` method. 
In this method you can pass named arguments for the channel attribute(s) that have to change.

### Example:
```py
@command()
async def update_channel():
    channel = await Channel.from_id(...)
    
    # the method to add
    await channel.edit(name="foo")

```

### CODE REVIEW

File: pincer/client.py [hunk 1]
```diff
@@ -5,20 +5,26 @@
 
 import logging
 from asyncio import iscoroutinefunction, run, ensure_future
+from importlib import import_module
 from inspect import isasyncgenfunction
 from typing import Optional, Any, Union, Dict, Tuple, List
 
+from pincer.objects import AppCommand
```
Comment (Arthurdw) [inline pincer/client.py:12]: Change this to relative import

Comment (Arthurdw) [reply pincer/client.py:12]: If its even used :?

**Review by Arthurdw — CHANGES_REQUESTED** [review]
Relative import if required

File: pincer/client.py (hunk 1 above)
Comment (Sigmanificient) [reply pincer/client.py:12]: used as a typehint at line 371 `to_remove: List[AppCommand] = []`

File: pincer/exceptions.py [hunk 2]
```diff
@@ -24,8 +24,49 @@ def __init__(self, specific: str):
 
 class NoExportMethod(PincerError):
     """
-    Exception which gets raised when an `export` method is expected but not
-    found in a module.
+    Exception which gets raised when an `export` method is expected but
```
Comment (Endercheif) [inline pincer/exceptions.py:27]: can this be changed from \`export\` to \`\`export\`\` so that it works with the docs

File: pincer/exceptions.py [hunk 3]
```diff
@@ -24,8 +24,49 @@ def __init__(self, specific: str):
...
+class CogAlreadyExists(CogError):
+    """
+    Exception which gets raised when a cog is already loaded, but is
+    trying to be reloaded!
+    """
+
+
+class NoValidSetupMethod(CogError):
+    """
+    Exception which gets raised when an `setup` function is expected but
```
Comment (Endercheif) [inline pincer/exceptions.py:54]: \`\`setup\`\`

File: pincer/middleware/ready.py [hunk 4]
```diff
@@ -23,7 +23,7 @@ async def on_ready_middleware(self, payload: GatewayDispatch):
         The data received from the ready event.
     """
     self.bot = User.from_dict(
-        {"_client": self, "_http": self.http,  **payload.data.get("user")}
+        {"_client": self, "_http": self.http, **payload.data.get("user")}
```
Comment (Endercheif) [inline pincer/middleware/ready.py:26]: If `payload.data.get("user")` returns None, then this would cause an error.

Comment (Arthurdw) [reply pincer/middleware/ready.py]: Resolved, now it will throw a `InvalidPayload`

File: pincer/exceptions.py (hunk 3 above)
Comment (Arthurdw) [reply pincer/exceptions.py]: Fixed

File: pincer/exceptions.py (hunk 2 above)
Comment (Arthurdw) [reply pincer/exceptions.py]: Fixed

File: pincer/client.py (hunk 1 above)
Comment (Arthurdw) [reply pincer/client.py]: Resolved with `TYPE_CHECKING`

**Review by Arthurdw — APPROVED** [review]
~ Ready for merge this time

**Pull request merged**